src/main/java/ivanov/springbootintro/controller/BookStoreController.java

```diff
@@ -54,16 +54,16 @@ public BookDto create(Authentication authentication,
     @PutMapping("/{id}")
     @Operation(summary = "Update book by id", description = "Update book by available id")
     public BookDto updateById(Authentication authentication,
-                              @RequestBody CreateBookRequestDto requestDto,
-                              @PathVariable Long id) {
+                              @RequestBody @Valid CreateBookRequestDto requestDto,
+                              @PathVariable @Valid Long id) {
         return bookService.updateById(requestDto, id);
     }
 
     @PreAuthorize("hasRole('ROLE_ADMIN')")
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
     @Operation(summary = "Delete book by id", description = "Delete book by available id")
-    public void deleteById(Authentication authentication, @PathVariable Long id) {
+    public void deleteById(Authentication authentication, @PathVariable @Valid Long id) {

```

Comment:
What does @Valid mean for Long argument?
```suggestion
    public void deleteById(Authentication authentication, @PathVariable Long id) {

```
Please recheck and remove all accurences